Sources/ApolloCodegenLib/ApolloCodegen.swift

```diff
@@ -1,32 +1,54 @@
 import Foundation
+import OrderedCollections
 
 // Only available on macOS
 #if os(macOS)
 
 /// A class to facilitate running code generation
 public class ApolloCodegen {
+  // MARK: Public
+
+  /// Errors that can occur during code generation.
   public enum Error: Swift.Error, LocalizedError {
-    case validationFailed(atLines: [String])
+    /// An error occured during validation of the GraphQL schema or operations.
+    case graphQLSourceValidationFailed(atLines: [String])
```

Comment:
I usually name my errors as nouns, not verbs. So, `graphQLSourceValidationFailure` is what I would call this.
But we don't have a style guide (and we really should), so this isn't wrong. Any opinion on which way you like this more? 

We really need to start putting together a style guide.

Reply:
I don't have a strong opinion either way. In many other parts of code I use the noun-naming convention so I'm happy to do the same here.